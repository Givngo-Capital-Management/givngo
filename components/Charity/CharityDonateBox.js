import React from 'react';
import { Button, Grid } from 'semantic-ui-react';
import styles from './CharityDonateBox.module.scss';

export default function CharityDonateBox() {
  return (
    <div className={styles.box}>
      <div className={styles.header}>
        <p>Donate now</p>
      </div>
      <div className={styles.content}>
        <Grid>
          <div className="equal width row">
            <div className="ten wide column">
              <h3 className="ui header">
                <div className="content">
                  Donate to Charity name
                  <div className="sub header">Charity short description</div>
                </div>
              </h3>
            </div>
            <div className="six wide column">
              <Button className="donate ui right floated" color="orange">
                Donate
              </Button>
            </div>
          </div>
        </Grid>

        <h3>Total raised: $132,619</h3>
      </div>
    </div>
  );
}
