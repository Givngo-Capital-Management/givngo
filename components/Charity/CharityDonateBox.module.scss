.box {
  display: block;
  border: 1px solid #ccc;
  margin-bottom: 24px; 
}

.header {
  padding: 16px;
  line-height: 2rem;
  background-color: rgb(241, 241, 241);
  font-size: 1.2rem;
  font-weight: 400;
}

.content {
  border-top: 1px solid #ccc;
  padding: 16px;
}