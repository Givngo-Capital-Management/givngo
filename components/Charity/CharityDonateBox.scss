.charity-donate-box {
  display: block;
  border: 1px solid #ccc;
  margin-bottom: 24px; 
}

.charity-donate-box-header {
  padding: 16px;
  line-height: 2rem;
  background-color: rgb(241, 241, 241);
  font-size: 1.2rem;
  font-weight: 400;
}

.charity-donate-box-content {
  border-top: 1px solid #ccc;
  padding: 16px;
}