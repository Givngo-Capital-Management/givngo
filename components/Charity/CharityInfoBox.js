import React from 'react';
import { Image, Button } from 'semantic-ui-react';
import styles from './CharityInfoBox.module.scss';

export default function CharityInfoBox() {
  return (
    <div className={styles.box}>
      <Image
        className={styles.image}
        src="http://via.placeholder.com/48x48"
        circular
      />
      <div className={styles.info}>
        <div className="charity-name">Charity Name</div>
        <span>80k followers</span>
      </div>
      <Button className={styles.follow} color="orange">
        Follow
      </Button>
      <div className={styles.description}>
        <p>
          Lorem Ipsum is simply dummy text of the printing and typesetting
          industry. Lorem Ipsum has been the industry&apos;s standard dummy text
          ever since the 1500s, when an unknown printer took a galley of type
          and scrambled it to make a type specimen book. It has survived not
          only five centuries, but also the leap into electronic typesetting,
          remaining essentially unchanged. It was popularised in the 1960s with
          the release of Letraset sheets containing Lorem Ipsum passages, and
          more recently with desktop publishing software like Aldus PageMaker
          including versions of Lorem Ipsum.
        </p>
      </div>
    </div>
  );
}
