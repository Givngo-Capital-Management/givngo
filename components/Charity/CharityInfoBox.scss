.charity-info-box {
  display: grid;
  grid: auto auto / 58px auto max-content;
  align-items: center;
  grid-row-gap: 16px;

  .charity-image {
    grid-row: 1 / 2;
    grid-column: 1 / 2;
  }

  .charity-info {
    grid-row: 1 / 2;
    grid-column: 2 / 3;

    .charity-name {
      font-weight: 600;
      cursor: pointer;
    }
    .charity-publication-date {
      font-size: 13px;
      color: #707070;
    }
  }

  .follow {
    grid-row: 1 / 2;
    grid-column: 3 / 4;
  }

  .charity-description {
    max-width: 615px;
    grid-row: 2 / 3;
    grid-column: 2 / 3;

    button {
      margin: 8px 0;
    }
  }
}