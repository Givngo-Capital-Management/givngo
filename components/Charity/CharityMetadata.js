import React from 'react';
import { Button, Divider, Icon } from 'semantic-ui-react';
import './CharityMetadata.scss';

export default function CharityMetadata() {
  return (
    <div className="charity-metadata">
      <h3>Video title</h3>
      <div className="charity-stats">
        <span>Mar 19, 2020</span>
        <div className="charity-actions">
          <Button basic icon labelPosition="left">
            <Icon name="share" />
            Share
          </Button>
        </div>
      </div>
      <Divider />
    </div>
  );
}
