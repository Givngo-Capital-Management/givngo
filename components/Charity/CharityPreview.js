import React from 'react';
import { Button, Image } from 'semantic-ui-react';
import './CharityPreview.scss';

export default function CharityPreview(props) {
  const { horizontal } = props;
  return (
    <div
      className={['charity-preview', horizontal ? 'horizontal' : null].join(
        ' ',
      )}
    >
      <div className="image-container">
        <Image src="http://via.placeholder.com/210x118" />
        <div className="time-label">
          <span>05:22</span>
        </div>
      </div>

      <div className="charity-info">
        <div className="semi-bold show-max-two-lines">Video title</div>
        <div className="charity-preview-metadata-container">
          <div className="charity-title">Charity name</div>
          <div>
            <span>November 2, 2020</span>
            <p>Short description</p>
          </div>
        </div>
        <Button className="mini compact ui orange basic smaller">
          <i className="heart outline icon" />
          Donate
        </Button>
        <a href="www.google.ca" className="charity-preview-link">
          <i className="star outline icon" />
          Follow
        </a>
        <a href="www.google.ca" className="charity-preview-link">
          <i className="share icon" />
          Share
        </a>
      </div>
    </div>
  );
}
