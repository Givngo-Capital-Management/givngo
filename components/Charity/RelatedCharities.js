import React from 'react';
import CharityPreview from './CharityPreview';
import './RelatedCharities.scss';

export default function RelatedCharities() {
  return (
    <div className="related-charities">
      <CharityPreview horizontal />
      <CharityPreview horizontal />
      <CharityPreview horizontal />
    </div>
  );
}
