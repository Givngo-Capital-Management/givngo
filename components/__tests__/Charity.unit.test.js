import React from 'react';
import { shallow } from 'enzyme';
import Charity from '../../pages/charity';

describe('Charity', () => {
  test('renders', () => {
    const wrapper = shallow(<Charity />);
    expect(wrapper).toMatchSnapshot();
  });
});
