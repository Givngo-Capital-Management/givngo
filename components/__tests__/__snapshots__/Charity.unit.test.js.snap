// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charity renders 1`] = `
<AppLayout>
  <Video
    id="374481280"
    title="breast friends"
  />
  <div
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Grid
      columns={2}
    >
      <GridColumn
        width={10}
      >
        <CharityMetadata />
        <CharityInfoBox />
      </GridColumn>
      <GridColumn
        floated="right"
        width={6}
      >
        <div>
          <CharityDonateBox />
          <RelatedCharities />
        </div>
      </GridColumn>
    </Grid>
  </div>
</AppLayout>
`;
